CPS-????/README.md

```diff
@@ -0,0 +1,137 @@
+---
+CPS: 24
+Title: Canonical CBOR Serialization Standard
+Category: Tools
+Status: Open
+Authors:
+    - Hinson Wong <wongkahinhinson@gmail.com>
+    - Tsz Wai Wu <tszwaiwu.96@gmail.com>
+Proposed Solutions: []
+Discussions:
+    - https://github.com/cardano-foundation/CIPs/pull/1109
+Created: 2025-10-29
+License: CC-BY-4.0
+---
+
+## Abstract
+
+There is no canonical CBOR serialization standard in Cardano. While this is a delibrate design choice initially, standardizing it has growing popularity in Cardano developer community as evidenced by developer meetups such as Cardano Builder Fest 2025 hosted in Vietnam. This CPS outlines the motivation of the growing concern of fragmented CBOR serialization patterns across in the community.
+
+## Problem
+
+<!-- A more elaborate description of the problem and its context. This section should explain what motivates the writing of the CPS document. -->
+
+The Cardano ledger accepts any valid CBOR encoding for transactions and on-chain data. While this flexibility was intentional to encourage ecosystem diversity, it has created significant interoperability challenges as the tooling landscape has matured. The same logical data can be encoded in multiple ways (map key ordering, integer encoding, definite vs. indefinite length, etc.), leading to different byte representations and transaction hashes.
+
+### Core Issues
+
+**Transaction Hash Instability**: When a transaction is passed between tools or wallets for signing, each may re-serialize it differently. Since transaction hashes are computed over CBOR bytes, logically identical transactions produce different hashes. This breaks:
+
+- Multi-signature workflows where each signer's wallet may re-serialize the transaction
+- Cross-tool transaction building where fee calculations depend on exact byte size
+
+**Script Inconsistencies**: Smart contracts suffer from unpredictable script hashes, reference script mismatches across tools. The same compiled script may produce different hashes depending on the library used to apply parameters or cbor serialize the script.
+
+**Development Friction**: Developers face increased testing burden across multiple libraries and wallets, library-specific test fixtures, vendor lock-in risks, and debugging challenges that require logical rather than byte-level comparison.
+
+### Ecosystem Impact
+
+The lack of standardization creates:
+
+- **Security risks**: Hard-to-diagnose bugs and complex audit requirements due to multiple serialization paths
+- **Community overhead**: High support burden in addressing serialization issues and maintaining multiple strategies
+- **Adoption barriers**: Unpredictable behavior discourages enterprise adoption and increases new developer friction
```

Comment:
I don't understand what is unpredictable in CBOR. It is a standard that 100% predictable.